src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import { Navigate } from 'react-router-dom';
+import { AuthProviderPropsType } from '../utils/helpers';
+import useAuth from '../hooks';
+
+const RequireAuthProvider: React.FC<AuthProviderPropsType> = ({ children }) => {
+  const { user, isLoading } = useAuth();
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  return user.userId ? (
+    children
+  ) : (
+    <Navigate to="/academy/login" />
+  );
```

Comment:
You can do it like this (using `useNavigate` hook it's better than using `Navigate` component:

```tsx
const navigate = useNavigate(); // from react-router-dom

if (!user) {
  navigate('/academy/login');
  return null;
}

return (
   children
);
```